secret_store/src/key_server_cluster/cluster.rs

```diff
@@ -1043,6 +1065,64 @@ impl ClusterClient for ClusterClientImpl {
 	}
 }
 
+impl EncryptionSessionWrapper {
+	pub fn new(cluster: &Arc<ClusterData>, session_id: SessionId, session: Arc<EncryptionSession>) -> Arc<Self> {
```

Comment:
typical style is to pass a `Weak` directly, this way it's more evident from the function signature that it's the caller's responsibility to keep the `Arc` alive.